tests/test_kv_store.c

```diff
@@ -232,6 +233,66 @@ static void test_kvstore_iterate(void **state)
 	release_test_data(test_iov);
 }
 
+
+static void test_kvstore_merge_op(void **state)
+{
+	struct iovec           test_iov[MAX_TEST_ENTRIES];
+	size_t                 data_size, index;
+	void *                 data;
+	kv_store_iter_t *      iter;
+	kv_store_value_flags_t flags;
+	sid_resource_t *       kv_store_res = NULL;
+	size_t                 test_data_size, kv_data_count;
+
+	kv_store_res = sid_resource_create(SID_RESOURCE_NO_PARENT,
+	                                   &sid_resource_type_kv_store,
+	                                   SID_RESOURCE_RESTRICT_WALK_UP,
+	                                   "testkvstore",
+	                                   &test_kv_store_res_params,
+	                                   SID_RESOURCE_PRIO_NORMAL,
+	                                   SID_RESOURCE_NO_SERVICE_LINKS);
+
+	test_data_size = add_test_data(test_iov);
+
+	assert_ptr_not_equal(kv_store_set_value(kv_store_res,
+	                                        MERGE_KEY,
+	                                        test_iov,
+	                                        MAX_TEST_ENTRIES,
+	                                        KV_STORE_VALUE_VECTOR,
+	                                        KV_STORE_VALUE_OP_MERGE,
+	                                        NULL,
+	                                        NULL),
```

Comment:
@prajnoha if I called the function supplying the last two parameters:

```
	assert_ptr_not_equal(kv_store_set_value(kv_store_res,
	                                        TEST_KEY,
	                                        test_iov,
	                                        MAX_TEST_ENTRIES,
	                                        KV_STORE_VALUE_VECTOR,
	                                        KV_STORE_VALUE_NO_OP,
	                                        test_kv_write_always,
	                                        &update_arg),
```
The data validation failed.  It only happens with OP_MERGE.

I'll look at why it fails.

Reply:
I was mistaken about how KV_STORE_VALUE_OP_MERGE works.  I see how it works now.